src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SingleExtensionEvalFunction.java

```diff
@@ -347,6 +401,32 @@ private void validateAllImportsAreGenerated(
     }
   }
 
+  /**
+   * @return usages with all information removed that does not influence the evaluation of the
+   *     extension, only the validation against imports that comes afterward.
+   */
+  private static ImmutableMap<ModuleKey, ModuleExtensionUsage> trimUsagesForEvaluation(
+      Map<ModuleKey, ModuleExtensionUsage> usages) {
+    return ImmutableMap.copyOf(
+        Maps.transformValues(
+            usages,
+            usage ->
+                // We start with the full usage and selectively remove information that does not
+                // influence the evaluation of the extension. Compared to explicitly copying over
+                // the parts that do, this preserves correctness in case new fields are added to
+                // ModuleExtensionUsage without updating this code.
+                usage.toBuilder()
+                    // Locations are only used for error reporting and thus didn't influence the
```

Comment:
s/didn't/don't/ ?

Reply:
I reformulated this slightly. I want to convey that while `Location`s can influence the result of evaluation in case of a failure (e.g. they show up in error messages), we can still trim them here.